src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -369,7 +367,7 @@ public new CSharpCompilation Clone()
                 this.AssemblyName,
                 _options,
                 this.ExternalReferences,
-                _previousSubmission,
+                this.PreviousSubmission,
                 this.SubmissionReturnType,
```

Comment:
Should the .ctor take a ```CSharpScriptCompilationInfo```, so we pass in ```this.ScriptCompilationInfo``` instead of creating a new instance each time?